queries: harmonize work type detection

maxlath opened the pull request:
hypothesis of this PR: using `VALUES` allows to more easily see the pattern than making those lenghtly `UNION`s

jum-s commented on server/data/wikidata/queries/editions_reverse_claims.coffee:
```diff
@@ -7,7 +7,10 @@ module.exports =
-      FILTER NOT EXISTS { ?item wdt:P31 wd:Q571 }
+      FILTER NOT EXISTS {
+        ?item wdt:P31 ?work_type
+        VALUES (?work_type) { (wd:Q571) (wd:Q47461344) (wd:Q7725634) (wd:Q2831984) (wd:Q1004) (wd:Q1760610) (wd:Q8274) } .
```
A usual pattern is to define a variable before using it, but here `?work_type` is defined after.

maxlath replied:
done in 04397e42

jum-s commented on server/data/wikidata/queries/works_reverse_claims.coffee:
```diff
@@ -5,26 +5,8 @@ module.exports =
-      # Filter-out entities tagged as both work and edition
-      FILTER NOT EXISTS { ?item wdt:P31 wd:Q3331189 }
+      ?item wdt:P31 ?work_type .
+      VALUES (?work_type) { (wd:Q571) (wd:Q47461344) (wd:Q7725634) (wd:Q2831984) (wd:Q1004) (wd:Q1760610) (wd:Q8274) } .
```
before ther was some comment on what Qids meant, which made it more readable

maxlath replied:
done in befb87f8

jum-s reviewed (changes requested):
unsure if its more readable, this `VALUES`  in SPARQL is new to us but it does not make it more readable for everyone. I would claim `UNION` is more explicit as its used in many SQL implementations

maxlath committed:
queries: harmonize work type detection

re-commit of 107218fc post-decaffeinate

maxlath committed:
queries: constraining ?work_type before refering to it

trying to make it more readable

re-commit of 04397e42 post-decaffeinate

maxlath committed:
queries: add comments

outside of the built query to avoid sending unnecessary long queries

re-commit of befb87f8 post-decaffeinate

maxlath commented:
I rewrote the commit for the post-decaffeinate era

maxlath commented:
on the `VALUES` version `UNION` syntax debate: I now generally prefer the `VALUES` syntax, as, when you are familiar with it, it is much more explicit on what is happening than when you have to inspect all `UNION` clauses to then figure out that it is the exact same pattern repeating. This would then allow `UNION` to be used as a way to highlight that there is a diversity of patterns being combined.
